services/embedding_service.py

```
@@ -3,54 +3,41 @@
-    # Expand mask for broadcasting
-    mask_expanded = attention_mask.unsqueeze(-1).expand(last_hidden.size()).float()
-    sum_hidden = (last_hidden * mask_expanded).sum(dim=1)
-    mask_sum = mask_expanded.sum(dim=1).clamp(min=1e-9)
-    embeddings = sum_hidden / mask_sum
-    return embeddings.cpu().tolist()
+    client = OpenAI(api_key=get_openai_api_key(), timeout=timeout, max_retries=max_retries)
+    for attempt in range(max_retries + 1):
+        try:
+            response = client.embeddings.create(input=texts, model=model)
+            return [item.embedding for item in response.data]
+        except AuthenticationError as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)